contractcourt/commit_sweep_resolver.go

```diff
@@ -252,18 +252,86 @@ func (c *commitSweepResolver) Resolve(_ bool) (ContractResolver, error) {
 
 		signDesc = c.commitResolution.SelfOutputSignDesc
 	)
+
+	// commitScriptTimeLockTaproot is a function that returns the timelock
+	// of a non-htlc commitment output.
+	//
+	// TODO(proofofkeags): move this
+	// NOTE FOR REVIEWERS: This function does not really belong here. That
+	// said, I don't know where it should go. If you have opinions, please
+	// comment.
+	commitScriptTimeLockTaproot := func(script []byte) fn.Option[int32] {
+		tok := txscript.MakeScriptTokenizer(0, script)
+		tok.Next() // PUSH <key>
+		if len(tok.Data()) != 32 {
+			return fn.None[int32]()
+		}
+		tok.Next() // CHECKSIG
+		if tok.Opcode() != txscript.OP_CHECKSIG {
+			return fn.None[int32]()
+		}
+
+		tok.Next() // PUSH <timelock>
+		var timelock int32
+		switch {
+		case tok.Opcode() == txscript.OP_0:
+			timelock = 0
+
+		case tok.Opcode() >= txscript.OP_1 &&
+			tok.Opcode() <= txscript.OP_16:
+
+			timelock = int32(tok.Opcode() - txscript.OP_1 + 1)
+
+		case tok.Opcode() == txscript.OP_DATA_1:
+			timelock = int32(tok.Data()[0])
+
+		case tok.Opcode() == txscript.OP_DATA_2:
+			timelock = int32(tok.Data()[0]) |
+				int32(tok.Data()[1])<<8
+
+		case tok.Opcode() == txscript.OP_DATA_3:
+			timelock = int32(tok.Data()[0]) |
+				int32(tok.Data()[1])<<8 |
+				int32(tok.Data()[2])<<16
+
+		case tok.Opcode() == txscript.OP_DATA_4:
+			timelock = int32(tok.Data()[0]) |
+				int32(tok.Data()[1])<<8 |
+				int32(tok.Data()[2])<<16 |
+				int32(tok.Data()[3])<<24
+		default:
+			// We don't cover any data pushes longer than 4 bytes as
+			// that would imply a timelock greater than the history
+			// of Bitcoin.
+			return fn.None[int32]()
+		}
+
+		tok.Next() // CSV
+		if tok.Opcode() != txscript.OP_CHECKSEQUENCEVERIFY {
+			return fn.None[int32]()
+		}
+
+		tok.Next() // DROP
+		if tok.Opcode() != txscript.OP_DROP {
+			return fn.None[int32]()
+		}
+
+		return fn.Some(timelock)
+	}
+
 	switch {
 	// For taproot channels, we'll know if this is the local commit based
-	// on the witness script. For local channels, the witness script has an
-	// OP_DROP value.
-	//
-	// TODO(roasbeef): revisit this after the script changes
-	//  * otherwise need to base off the key in script or the CSV value
-	//  (script num encode)
+	// on the timelock value. For remote commitment transactions, the
```

Comment:
Tbh I think all this info should already been provided by the resolutions (in this case `CommitOutputResolution`). Then from these resolutions, we can easily construct the resolvers without having to guess the witness type again. For instance, 
- For local FC, `LogContractResolutions` is called inside `<-c.cfg.ChainEvents.LocalUnilateralClosure`.
- For remote FC, `LogContractResolutions` is called inside `<-c.cfg.ChainEvents.RemoteUnilateralClosure`.

Reply:
Yeah this is another route as well, we already have the context. I think for this route though, we'd need to add additional state. 

Here I propose just looking at the payment base points: https://github.com/lightningnetwork/lnd/pull/8879#discussion_r1663359119

Reply:
I went with supplementing the local ChannelConfig